### PR TITLE
Modernize C++ in llapperanceutil

### PR DESCRIPTION
**Pull request by Nicky-D**
- Add override and remove /*virtual*/ comment
- NULL to nullptr
- FALSE to false
- TRUE to true

### CODE REVIEW

File: indra/llappearanceutility/llbakingtexture.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@
 #include "llgltexture.h"
 #include "llimage.h"
 
-static const bool USE_MIP_MAPS = TRUE;
+static const bool USE_MIP_MAPS = true;
```
Comment (Ansariel) [inline indra/llappearanceutility/llbakingtexture.cpp:35]: Also a candidate for constexpr

File: indra/llappearanceutility/llbakingwindow.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ LLBakingWindow::LLBakingWindow(S32 width, S32 height)
     const bool NO_CLEAR_BG = FALSE;
```
Comment (Ansariel) [inline indra/llappearanceutility/llbakingwindow.cpp:44]: More fake booleans here that are also candidates for constexpr

File: indra/llappearanceutility/llbakingtexture.cpp (hunk 1 above)
Comment (Nicky-D) [reply indra/llappearanceutility/llbakingtexture.cpp:35]: I had actually forgotten to git add some files ...
Thanks for pointing that out, those booleans in a fat suite are also replaced now

File: indra/llappearanceutility/llbakingavatar.h [hunk 3]
```diff
@@ -66,43 +66,41 @@ class LLBakingAvatar : public LLAvatarAppearance
...
+    void            getGround(const LLVector3 &inPos, LLVector3 &outPos, LLVector3 &outNorm) override;
+    F32             getPixelArea() const override;
+    LLVector3d      getPosGlobalFromAgent(const LLVector3 &position) override;
+    LLVector3       getPosAgentFromGlobal(const LLVector3d &position) override;
 
     //--------------------------------------------------------------------
     // LLAvatarAppearance interface
     //--------------------------------------------------------------------
 public:
-    /*virtual*/ void    bodySizeChanged();
```
Comment (nat-goodspeed) [inline indra/llappearanceutility/llbakingavatar.h:86]: In this block of `/*virtual*/` -> `override` changes, it's odd to see a couple method declarations vanish: `bodySizeChanged()` and `isUsingServerBakes()`. I'm not objecting, but I'd like a word of explanation in this PR.

File: indra/llappearanceutility/llbakingavatar.h [hunk 4]
```diff
@@ -114,9 +112,8 @@ class LLBakingAvatar : public LLAvatarAppearance
  **                    STATE
  **/
 public:
-    /*virtual*/ bool    isSelf() const { return true; }
-    /*virtual*/ bool    isValid() const { return TRUE; }
-    /*virtual*/ bool    isUsingBakedTextures() const { return TRUE; }
```
Comment (nat-goodspeed) [inline indra/llappearanceutility/llbakingavatar.h:119]: another vanishing method

**Review by nat-goodspeed — APPROVED** [review]

File: indra/llappearanceutility/llbakingavatar.h (hunk 4 above)
Comment (Nicky-D) [reply indra/llappearanceutility/llbakingavatar.h:119]: Those had vanished from the base long ago, effectively doing nothing (and generating override errors: "This method does not override something") 
From my understanding was this all "must have this to inherit, so dummy override this" methods.

You are right I should have included the reasoning in the fimmut comment. I will update accordingly

**Review by marchcat — APPROVED** [review]

**Comment by rohvani** [comment]
Thank you so much for your contribution, @Nicky-D! We really appreciate your effort. In the future, could you please reach out before submitting a PR against an active feature branch?

You can get in touch with us in-world via the weekly user groups, by directly IMing us in-world, or by emailing us at our Linden Lab addresses (you can reach me at [pepper@lindenlab.com](mailto:pepper@lindenlab.com)).

Thanks again!

**Pull request merged**